pkg/gcv/configs/config.go

```diff
@@ -80,44 +80,44 @@ func (c *UnclassifiedConfig) AsInterface() (interface{}, error) {
 // Returns:
 //   *Constraint: only set if valid constraint
 //   bool: (always set) if this is a constraint
-func asConstraint(data *UnclassifiedConfig) (*Constraint, bool, error) {
+func asConstraint(data *UnclassifiedConfig) (*Constraint, error) {
 	// There is no validation matching this constraint to the template here that happens after
 	// basic parsing has happened when we have more context.
 	if data.Group != validConstraintGroup {
-		return nil, false, nil // group is not a valid group
+		return nil, fmt.Errorf("group expected to be %s not %s", validConstraintGroup, data.Group)
 	}
 	if data.Kind == "ConstraintTemplate" {
-		return nil, true, fmt.Errorf("kind should not be ConstraintTemplate")
+		return nil, fmt.Errorf("kind should not be ConstraintTemplate")
```

Comment:
"kind should be ..."? also let's mention the bad kind?

Reply:
The kind can be anything except this. Since it correlates to the user defined kind in the template.